gpg to gpg2:
Arch's "gnupg" package is this distro's "gnupg2"
This patches all references to the gpg command to use
gpg2 instead.

diff -Naur pacman-5.2.1.orig/scripts/pacman-key.sh.in pacman-5.2.1/scripts/pacman-key.sh.in
--- scripts/pacman-key.sh.in	2019-10-24 06:29:08.000000000 -0700
+++ scripts/pacman-key.sh.in	2020-05-24 02:30:26.838797182 -0700
@@ -51,6 +51,7 @@
 UPDATEDB=0
 USE_COLOR='y'
 VERIFY=0
+GPG_COMMAND='gpg2'

 usage() {
 	printf "pacman-key (pacman) %s\n" ${myver}
@@ -591,8 +592,8 @@
 	unset ALL_OFF BOLD BLUE GREEN RED YELLOW
 fi

-if ! type -p gpg >/dev/null; then
-	error "$(gettext "Cannot find the %s binary required for all %s operations.")" "gpg" "pacman-key"
+if ! type -p ${GPG_COMMAND} >/dev/null; then
+	error "$(gettext "Cannot find the %s binary required for all %s operations.")" "${GPG_COMMAND}" "pacman-key"
 	exit 1
 fi

@@ -611,7 +612,7 @@
 # file, falling back on a hard default
 PACMAN_KEYRING_DIR=${PACMAN_KEYRING_DIR:-$(pacman-conf --config="$CONFIG" gpgdir)}

-GPG_PACMAN=(gpg --homedir "${PACMAN_KEYRING_DIR}" --no-permission-warning)
+GPG_PACMAN=(${GPG_COMMAND} --homedir "${PACMAN_KEYRING_DIR}" --no-permission-warning)
 if [[ -n ${KEYSERVER} ]]; then
 	GPG_PACMAN+=(--keyserver "${KEYSERVER}")
 fi


